### ISSUE TITLE
Setting specific IP of new VM instance is not properly propagated to the backend

### ISSUE THREAD

**Issue by dredknight** [apache/cloudstack#7175]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.17.2
~~~

##### CONFIGURATION

Advanced Network Zone with SG enabled. Deployed on XCP-NG with Linux bridge.

##### OS / ENVIRONMENT

Tested on:
-  QubeOS with Firefox exported from Debian 11.6 VM. Firefox version is: Firefox ESR 102.5.0esr (64-bit)
- Debian 11 OS with Google Chrome - Version 110.0.5481.52 (Official Build) snap (64-bit)

##### STEPS TO REPRODUCE

~~~

1. Open the GUI and go to Compute
2. Create new instance 
3. Choose existing template and network
4. Set valid IP address  in from the chosen network range (in the current example i have chosen 172.20.0.45)
5. Create the new instance
~~~

##### EXPECTED RESULTS

~~~
VM to be created with the ip selected
~~~

##### ACTUAL RESULTS

~~~
In the log it can be seen that the IP the backend gets is cut by 1 character. Instead of 172.20.0.45 it gets 172.20.0.4.
This happens for any IP selected. If 172.0.20.93 is set than the log will prompt 172.20.0.9

2023-02-07 08:23:37,316 WARN  [c.c.n.IpAddressManagerImpl] (qtp1648001170-964145:ctx-bd0c7c8d ctx-eab4d507) (logid:241533f1) Unable to get ip address in  zone id=1, network id=406: requested ip 172.20.0.4 is not available

Full log:
https://gist.github.com/dredknight/da192544c5e7a4b83bcf30d6f558742e
~~~

**Comment by weizhouapache** [comment]
@dredknight 
This looks same as #6974 , which has been fixed by #6994

**Comment by dredknight** [comment]
Agreed!
